targets/csharp/source/source/Uunit/PluginManagerTest.cs

```diff
@@ -0,0 +1,353 @@
+            }
+            catch (Exception e)
+            {
+                testContext.EndTest(UUnitFinishState.FAILED, e.ToString());
+            }
+            finally
+            {
+                // Restore the original plugin
+                PluginManager.SetPlugin(playFabSerializer, PluginContract.PlayFab_Serializer);
+            }
+        }
+
+        /// <summary>
+        /// CLIENT AND SERVER API
+        /// Test that plugin manager can set and return mupltiple plugins per contract.
+        /// </summary>
+        [UUnitTest]
+        public void PluginManagerMultiplePluginsPerContract(UUnitTestContext testContext)
+        {
+            const string customTransportName1 = "Custom transport client 1";
+            const string customTransportName2 = "Custom transport client 2";
+
+            var playFabTransport = PluginManager.GetPlugin(PluginContract.PlayFab_Transport) as ITransportPlugin;
+            var customTransport1 = new CustomTransportPlugin() { Name = customTransportName1 };
+            var customTransport2 = new CustomTransportPlugin() { Name = customTransportName2 };
+
+            // Set a custom plugins
+            PluginManager.SetPlugin(customTransport1, PluginContract.PlayFab_Transport, customTransportName1);
+            PluginManager.SetPlugin(customTransport2, PluginContract.PlayFab_Transport, customTransportName2);
+
+            // Verify 
+            var transport = PluginManager.GetPlugin(PluginContract.PlayFab_Transport) as ITransportPlugin;
+            testContext.True(object.ReferenceEquals(transport, playFabTransport)); // the default one is still the same
+            var transport1 = PluginManager.GetPlugin(PluginContract.PlayFab_Transport, customTransportName1) as ITransportPlugin;
+            testContext.True(object.ReferenceEquals(transport1, customTransport1));
+            var transport2 = PluginManager.GetPlugin(PluginContract.PlayFab_Transport, customTransportName2) as ITransportPlugin;
+            testContext.True(object.ReferenceEquals(transport2, customTransport2));
+            testContext.EndTest(UUnitFinishState.PASSED, null);
+        }
+
+        /// <summary>
+        /// CLIENT AND SERVER API
+        /// Test that SDK can use a custom serializer plugin.
+        /// </summary>
+        [UUnitTest]
+        public void SdkCustomPluginSerializer(UUnitTestContext testContext)
+        {
+            var playFabSerializer = PluginManager.GetPlugin(PluginContract.PlayFab_Serializer) as ISerializerPlugin;
+            var customSerializer = new CustomSerializerPlugin();
+            try
+            {
+                // Set a custom serializer plugin
+                PluginManager.SetPlugin(customSerializer, PluginContract.PlayFab_Serializer);
+
+                // Call some PlayFab API 
+                var getRequest = new GetUserDataRequest();
+                var getDataTask = PlayFabClientAPI.GetUserDataAsync(getRequest, null, extraHeaders);
```

Comment:
I still feel like it's unnecessary to make PlayFab API calls as part of testing PluginManager.
It puts too much on the test, and too many disconnected things can cause the test to fail.